File: apps/macos/PasteSheets/Domain/UseCases/SettingsUseCase.swift
import Foundation

final class SettingsUseCase {
    private let repo: SettingsRepository
    private let mouseEdgeService: MouseEdgeService
    private let autoStartService: AutoStartService

    init(repo: SettingsRepository,
         mouseEdgeService: MouseEdgeService,
         autoStartService: AutoStartService) {
        self.repo = repo
        self.mouseEdgeService = mouseEdgeService
        self.autoStartService = autoStartService
    }

    func getSetting(key: String) throws -> String? {
        try repo.getSetting(key: key)
    }

    func setSetting(key: String, value: String) throws {
        try repo.setSetting(key: key, value: value)

        if key == "mouse_edge_enabled" {
            mouseEdgeService.setEnabled(value == "true")
        }
    }

    func setAutoStart(enabled: Bool) throws {
        if enabled {
            try autoStartService.enable()
        } else {
            try autoStartService.disable()
        }
        try repo.setSetting(key: "auto_start", value: enabled ? "true" : "false")
    }

    func isAutoStartEnabled() -> Bool {
        autoStartService.isEnabled()
    }
}

File: apps/windows/PasteSheet/Domain/UseCases/SettingsUseCase.cs
using PasteSheet.Domain.Repositories;
using PasteSheet.Services;

namespace PasteSheet.Domain.UseCases;

public sealed class SettingsUseCase
{
    private readonly ISettingsRepository _repo;
    private readonly MouseEdgeService _mouseEdgeService;
    private readonly AutoStartService _autoStartService;

    public SettingsUseCase(ISettingsRepository repo, MouseEdgeService mouseEdgeService, AutoStartService autoStartService)
    {
        _repo = repo;
        _mouseEdgeService = mouseEdgeService;
        _autoStartService = autoStartService;
    }

    public string? GetSetting(string key) => _repo.GetSetting(key);

    public void SetSetting(string key, string value)
    {
        _repo.SetSetting(key, value);
        if (key == "mouse_edge_enabled")
            _mouseEdgeService.SetEnabled(value == "true");
    }

    public void SetAutoStart(bool enabled)
    {
        if (enabled) _autoStartService.Enable();
        else _autoStartService.Disable();
        _repo.SetSetting("auto_start", enabled ? "true" : "false");
    }

    public bool IsAutoStartEnabled() => _autoStartService.IsEnabled();
}
